# Description of `Test centers` table

In [1]:
import pandas as pd
import seaborn as sns

In [2]:
df = pd.read_csv('../tables_creation/final_tables/test_centers_edrpou.csv')
df

ptname  KOATUU_2020  \
0                Вище професійне училище №11 м. Вінниці    510100000   
1     Відокремлений структурний підрозділ "Вінницьки...    510100000   
2     Відокремлений структурний підрозділ Вінницький...    510100000   
3     Вінницький коледж Національного університету х...    510100000   
4     Вінницький коледж будівництва і архітектури Ки...    510100000   
...                                                 ...          ...   
3353  Школа №199 І-ІІІ ступенів Шевченківського райо...   8039100000   
3354  Школа №25 І-ІІІ ступенів Шевченківського район...   8039100000   
3355  Школа №70 І-ІІІ ступенів Шевченківського район...   8039100000   
3356  Школа №95 І-ІІІ ступенів Шевченківського район...   8039100000   
3357  школа І-ІІІ ступенів №175 ім.В Марченка Шевчен...   8039100000   

                                year    EDRPOU  
0           [2018, 2019, 2020, 2021]   3065891  
1                             [2021]  35725833  
2                             [2021]  38814461  
3                 [2018, 2019, 2020]  35725833  
4     [2016, 2017, 2018, 2019, 2020]  38814461  
...                              ...       ...  
3353                          [2016]  22881923  
3354        [2018, 2019, 2020, 2021]  22880786  
3355                          [2016]  22881828  
3356                          [2017]  26125905  
3357                          [2021]  26125940  

[3358 rows x 4 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3358 entries, 0 to 3357
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   ptname       3358 non-null   object
 1   KOATUU_2020  3358 non-null   int64 
 2   year         3358 non-null   object
 3   EDRPOU       3358 non-null   int64 
dtypes: int64(2), object(2)
memory usage: 105.1+ KB


In [4]:
# unique value count in ptname, KOATUU_2020, EDRPOU
print('Unique length in ptname:\t', len(df['ptname'].unique()))
print('Unique length in KOATUU_2020:\t', len(df['KOATUU_2020'].unique()))
print('Unique length in EDRPOU:\t', len(df['EDRPOU'].unique()))

Unique length in ptname:	 3339
Unique length in KOATUU_2020:	 507
Unique length in EDRPOU:	 2111


Features:
- ptname (name of the test center): 3339 unique test centeres names, where students can pass EIE or NMT. It can be duplicates of test centers names due to mistakes in the data, better to use only EDRPOU.
- KOATUU_2020 (KOATUU code): 507 unique location code, where located test center(a unique digital code for each administrative-territorial entity of Ukraine, used from January 1998 till January 2021): territory, city area, other countries city. The first two digits signify the region, the next two represent the district, and the following two denote the city or municipality.
- EDRPOU (unique code of the legal entity or physical person): 2110 unique codes of legal entities, who are owners of test centers(unique test centers code).